src/python-common/ceph/deployment/inventory.py

```diff
@@ -15,7 +15,8 @@ class Devices(object):
 
     def __init__(self, devices):
         # type: (List[Device]) -> None
-        self.devices = devices  # type: List[Device]
+        # sort devices by path name so ordering is consistent
+        self.devices: List[Device] = sorted(devices, key=lambda d: d.path if d.path else '')
```

Comment:
```
________________________ TestHosts.test_get_inventories ________________________

self = <dashboard.tests.test_host.TestHosts testMethod=test_get_inventories>
mock_get_device_osd_map = <MagicMock name='get_device_osd_map' id='140422417343984'>
mock_str_to_bool = <MagicMock name='str_to_bool' id='140422417334720'>

    @mock.patch('dashboard.controllers.host.str_to_bool')
    @mock.patch('dashboard.controllers.host.get_device_osd_map')
    def test_get_inventories(self, mock_get_device_osd_map, mock_str_to_bool):
        mock_get_device_osd_map.return_value = {
            'host-0': {
                'nvme0n1': [1, 2],
                'sdb': [1],
                'sdc': [2]
            },
            'host-1': {
                'sdb': [3]
            }
        }
        inventory = [
            {
                'name': 'host-0',
                'addr': '1.2.3.4',
                'devices': [
                    {'path': 'nvme0n1'},
                    {'path': '/dev/sdb'},
                    {'path': '/dev/sdc'},
                ]
            },
            {
                'name': 'host-1',
                'addr': '1.2.3.5',
                'devices': [
                    {'path': '/dev/sda'},
                    {'path': 'sdb'},
                ]
            }
        ]
    
        with patch_orch(True, inventory=inventory) as orch_client:
            mock_str_to_bool.return_value = True
    
            hosts = ['host-0', 'host-1']
            inventories = get_inventories(hosts, 'true')
            mock_str_to_bool.assert_called_with('true')
            orch_client.inventory.list.assert_called_once_with(hosts=hosts, refresh=True)
            self.assertEqual(len(inventories), 2)
            host0 = inventories[0]
            self.assertEqual(host0['name'], 'host-0')
            self.assertEqual(host0['addr'], '1.2.3.4')
>           self.assertEqual(host0['devices'][0]['osd_ids'], [1, 2])
E           AssertionError: Lists differ: [1] != [1, 2]
E           
E           Second list contains 1 additional elements.
E           First extra element 1:
E           2
E           
E           - [1]
E           + [1, 2]

tests/test_host.py:502: AssertionError
```
looks related to this patch. Will look into this.

Reply:
swapped the ordering of the checks in this test since the devices should now be sorted by path name in the inventory and it passed locally